bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/IntermediateCatchEventValidator.java

```diff
@@ -37,16 +40,52 @@ public void validate(
       IntermediateCatchEvent element, ValidationResultCollector validationResultCollector) {
 
     final Collection<EventDefinition> eventDefinitions = element.getEventDefinitions();
-    final Optional<EventDefinition> messageEventDefinition =
-        eventDefinitions.stream().filter(e -> e instanceof MessageEventDefinition).findFirst();
 
-    if (!messageEventDefinition.isPresent()) {
-      validationResultCollector.addError(0, "Must have a message event definition");
+    if (eventDefinitions.isEmpty()) {
+      validationResultCollector.addError(0, "Must have a message or timer event definition");
 
+    } else if (eventDefinitions.size() > 1) {
+      validationResultCollector.addError(0, "Must have exactly one event definition");
+
+    } else {
+      final EventDefinition eventDefinition = eventDefinitions.iterator().next();
+
+      if (eventDefinition instanceof MessageEventDefinition) {
+        validateMessageEventDefinition(
+            (MessageEventDefinition) eventDefinition, validationResultCollector);
+
+      } else if (eventDefinition instanceof TimerEventDefinition) {
+        validateTimerEventDefinition(
+            validationResultCollector, (TimerEventDefinition) eventDefinition);
+
+      } else {
+        validationResultCollector.addError(0, "Must have a message or timer event definition");
+      }
+    }
+  }
+
+  private static void validateMessageEventDefinition(
+      final MessageEventDefinition messageEventDefinition,
+      ValidationResultCollector validationResultCollector) {
+
+    final Message message = messageEventDefinition.getMessage();
+    if (message == null) {
+      validationResultCollector.addError(0, "Must reference a message");
+    }
+  }
+
+  private void validateTimerEventDefinition(
+      ValidationResultCollector validationResultCollector,
+      final TimerEventDefinition timerEventDefinition) {
+
+    final TimeDuration timeDuration = timerEventDefinition.getTimeDuration();
+    if (timeDuration == null) {
```

Comment:
At some point, it may be possible to set a `timeDate`. But a `timeCycle` is not possible because the intermediate catch event can be triggered only once.